File: src/tools/dirb/dirb.py
import argparse
import requests

def check_dir_exists(url):
    response = requests.head(url)
    return response.status_code in (200, 301)

def main():
    parser = argparse.ArgumentParser(description='Directory buster')
    parser.add_argument('base_url', help='Base URL for directory scanning')
    args = parser.parse_args()

    base_url = args.base_url

    try:
        with open('wordlist.txt', 'r') as file:
            directories = file.read().splitlines()
    except FileNotFoundError as e:
        print(f"Error opening wordlist.txt: {e}")
        return

    print("Directory buster started...")

    for dir in directories:
        url = f"{base_url}/{dir}"
        if check_dir_exists(url):
            print(f"[GET] Directory found: {url}")

if __name__ == "__main__":
    main()

File: src/tools/dirb/dirb.go
package main

import (
	"bufio"
	"fmt"
	"net/http"
	"os"
)

func checkDirExists(url string) bool {
	response, err := http.Get(url)
	if err != nil {
		return false
	}
	defer response.Body.Close()

	// Check if the HTTP status code is 200 (OK) or 301 (Moved Permanently)
	if response.StatusCode == http.StatusOK || response.StatusCode == http.StatusMovedPermanently {
		return true
	}
	return false
}

func main() {
	if len(os.Args) < 2 {
		return
	}

	baseURL := os.Args[1]

	// Read the wordlist from the file
	file, err := os.Open("lists/*.txt")
	if err != nil {
		fmt.Println("Error opening wordlist.txt:", err)
		return
	}
	defer file.Close()

	scanner := bufio.NewScanner(file)
	var directories []string

	// Store each line (word) from the file in the directories slice
	for scanner.Scan() {
		directories = append(directories, scanner.Text())
	}

	if err := scanner.Err(); err != nil {
		fmt.Println("Error reading wordlist.txt:", err)
		return
	}

	fmt.Println("Directory buster started...")

	// Iterate over the directories and make requests
	for _, dir := range directories {
		url := baseURL + "/" + dir
		if checkDirExists(url) {
			fmt.Printf("[%s] Directory found: %s\n", http.MethodGet, url)
		}
	}
}